tutorials/preprocessing/plot_40_artifact_correction_ica.py

```diff
@@ -1,307 +1,549 @@
+# independent source signals from a set of recordings in which the source
+# signals were mixed together in unknown ratios. A common example of this is
+# the problem of `blind source separation`_: with 3 musical instruments playing
+# in the same room, and 3 microphones recording the performance (each picking
+# up all 3 instruments, but at varying levels), can you somehow "unmix" the
+# signals recorded by the 3 microphones so that you end up with a separate
+# "recording" isolating the sound of each instrument?
+#
+# It is not hard to see how this analogy applies to EEG/MEG analysis: there are
+# many "microphones" (sensor channels) simultaneously recording many
+# "instruments" (blinks, heartbeats, activity in different areas of the brain,
+# muscular activity from jaw clenching or swallowing, etc). As long as these
+# various source signals are `statistically independent`_ and non-gaussian, it
+# is usually possible to separate the sources using ICA, and then re-construct
+# the sensor signals after excluding the sources that are unwanted.
+#
+#
+# ICA in MNE-Python
+# ~~~~~~~~~~~~~~~~~
+#
+# MNE-Python implements three different ICA algorithms: ``fastica`` (the
+# default), ``picard``, and ``infomax``. FastICA and Infomax are both in fairly
+# widespread use; Picard is a newer (2017) algorithm that is expected to
+# converge faster than FastICA and Infomax, and is more robust than other
+# algorithms in cases where the sources are not completely independent, which
+# typically happens with real EEG/MEG data. See [1]_ for more information.
+#
+# The ICA interface in MNE-Python mirrors the standard interface in
+# `scikit-learn`_: some general parameters are specified when creating an
+# :class:`~mne.preprocessing.ICA` object, then the
+# :class:`~mne.preprocessing.ICA` object is fit to the data using its
+# :meth:`~mne.preprocessing.ICA.fit` method. The results of the fitting are
+# added to the :class:`~mne.preprocessing.ICA` object as attributes that end in
+# an underscore (``_``), such as ``ica.mixing_matrix_`` and
+# ``ica.unmixing_matrix_``. After fitting, the ICA component(s) that you want
+# to remove must be chosen, and the ICA fit must then be applied to the
+# :class:`~mne.io.Raw` or :class:`~mne.Epochs` object using the
+# :class:`~mne.preprocessing.ICA` object's :meth:`~mne.preprocessing.ICA.apply`
+# method.
+#
+# As is typically done with ICA, the data are first
+# scaled to unit variance and whitened using principal components analysis
+# (PCA) before performing the ICA decomposition. You can impose a
+# dimensionality reduction at this step by specifying ``max_pca_components``.
+# From those retained Principal Components (PCs), the first ``n_components``
+# are then passed to the ICA algorithm (``n_components`` may be an integer
+# number of components to use, or a fraction of explained variance that used
+# components should capture). Then, when reconstructing the signal after
```

Comment:
I'm sorry I did not participate in earlier discussions about this. But could you explain why there are two parameters and not one for `max_pca_components` + `n_components` ?